### DISCUSSION TITLE
Xero Integration

### DISCUSSION THREAD

**Discussion by bitcoinbrisbane** [btcpayserver/btcpayserver#983] [Plugins & integrations]
Allow BTC Pay server to push invoices into Xero.

- As a business owner,  
When I create an invoice in BTC pay server, I want to see it added to Xero,   
So that I don't have to double enter my information

- As a business owner,
When I create an invoice is paid by crypto via BTC pay server, I want to see it marked as paid in Xero,
So that I don't have to double enter my information

**Comment by Kukks** [comment]
I think this would make more sense as a separate plugin to BTCPay like the quickbooks plugin in the docker repo

**Comment by astupidmoose** [comment]
Does Xero have BTC in their accounting software? Last time I checked, which was recently it was not an option.

**Comment by bolatovumar** [comment]
@astupidmoose doesn't look like they do.

See here: https://community.xero.com/business/discussion/305641
But looks like people are using it with some other tools to make it work: https://medium.com/cryptfolio/tracking-cryptocurrency-payments-in-xero-5d0018056dca

**Comment by bitcoinbrisbane** [comment]
> Does Xero have BTC in their accounting software? Last time I checked, which was recently it was not an option.

Could be via FIAT?  Raise invoice in FIAT, paid in BTC, mark invoice as paid?

**Comment by ksdhans** [comment]
Just over two years later, and Xero still doesn't have cryptocurrency support. Nor do they have plans to implement it in the next year. 

However, in multiple countries (including Australia and NZ), crypto is treated similarly to shares or other assets. So, you can create an account in your local currency, and use the current exchange rate to handle transactions. You'll also need to adjust for currency gains/losses whenever spending. Nowhere near as good as having full crypto support, but it is doable (see the links in bolatovumar's comment above). 

A Xero plugin for BTCPay would be useful, because it could automate the process outlined above.

**Reply by bitcoinbrisbane** [reply]
Yep, seems the issue is the decimals in Xero too.  They only support 2 decimals, so everything would need to be in sats etc.

**Reply by Tristan-van-de-Beld** [reply]
@bitcoinbrisbane 

The use case I have is I want to create Invoices in Xero or BTC pay server and for it to cross sync. After talking to my accountant this would not be denominated in BTC but rather local fiat but paid in BTC at market price.

I think starting with this, where the whole Xero side is denominated in Fiat would provide the majority of the needed functionality while maintaining integration simplicity. 

Let me know your thoughts please. I have also worked a fair bit on the Xero API over the years.

**Reply by ksdhans** [reply]
@Tristan-van-de-Beld 
That sounds like roughly what I'm doing (manually). Bitcoin sales go to a bitcoin account in NZD. I have to adjust for currency gains when I spend/convert bitcoin, which is tedious to do manually.
